src/DataProvider.js

```diff
@@ -67,22 +68,31 @@ export default class DataProvider {
     }, this.polling())
   }
 
-  fetch() {
+  async fetch() {
     if (this.canceled()) {
       return null
     }
     if (this.timer) {
       clearTimeout(this.timer)
     }
 
-    return this.getData()
-      .then((data) => {
-        if (!this.canceled()) {
-          this.loaded = true
-          this.onData(data)
-        }
-      }).catch((e) => {
-        console.error(e) // eslint-disable-line no-console
-      }).then(() => this.scheduleNextFetch())
+    let data
+    if (cfg.ignoreGetDataErrors) {
+      try {
+        data = await this.getData()
+      } catch (error) {
+        return Promise.reject()
```

Comment:
Hmm, but if you return the rejected promise here, the next fetch is not scheduled - is it?

Reply:
Ahh right, good catch :+1:

Reply:
I've been thinking about this, whether we should `return` here at all. Maybe the purpose of the `ignoreGetDataErrors` flag should be just that - catch the error, maybe set the `data` to `null`, and continue execution, i.e. execture `onData()` as well...
What do you think?

Reply:
After a discussion with Tomas, I've reimplemented this using new fetchWithRetry() function.